### ISSUE TITLE
Calculation of mean for calendar days

### ISSUE THREAD

**Issue by temenoskrpavithra** [influxdata/influxdb#8183]
I'm currently exploring InfluxDB(V1.2) now for setting up metrics system. 
I tried calculating mean of few values to understand the working.
My points are 
time                    distinct
----                    --------
1486112400000000000     3
1486198800000000000     0
1486285200000000000     8
1486357200000000000     5
1486371600000000000     2
1486371600000000000     4
1486371600000000000     5
1486458000000000000     2

It starts only from Feb-2-2017
Setting up the date range from 1-Jan-2017 to 28-Feb-2017, with groupby time(30d), i find the mean calculated fine, but it is against the date 24-Jan-2017.
> SELECT mean("VAL") FROM "P1" WHERE time >=1483228800000 AND time >=1486598400000 GROUP BY time(15d) fill(none)
name: P1
time                    mean
----                    ----
1485216000000000000     3.5555555555555554

> SELECT mean("VAL") FROM "P1" WHERE time >=1483228800000 AND time >=1486598400000 GROUP BY time(30d) fill(none)
name: P1
time                    mean
----                    ----
1485216000000000000     3.5555555555555554

Basically, **my requirement is to calculate mean for 30days ( for calendar days )**
In this case, for Jan, I should see 0 and Feb 3.5
Why does the mean specifically point to Jan-24th ? I have no points in Jan . If i plot this in grafana, the mean shown against the dates show misleading data.

What is the way to show mean of calendar months on the graph...﻿

I'm sorry for posting this here, but i haven't received any help on the community. I have implemented a workaround to capture mean for the month by introducing tags. However, I don't find it a neat solution.
Any suggestions around calculating mean for monthly period would be very much appreciated.
